File: Engine/src/main/java/org/etieskrill/engine/config/GLContextConfig.java
package org.etieskrill.engine.config;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class GLContextConfig {

    //TODO handle thread context transfer

    public static final ThreadLocal<GLContextConfig> CONFIG = ThreadLocal.withInitial(() -> {
        throw new IllegalStateException("Tried to access config for uninitialised GL context");
    });

    private final int _maxTextureUnits;

    public static int getMaxTextureUnits() {
        return CONFIG.get().get_maxTextureUnits();
    }

}

File: Engine/src/main/java/org/etieskrill/engine/config/GLContextConfig.kt
package org.etieskrill.engine.config;

data class GLContextConfig(
    val maxTextureUnits: Int
) {

    //TODO handle thread context transfer

    companion object {
        val CONFIG: ThreadLocal<GLContextConfig> = ThreadLocal.withInitial {
            throw IllegalStateException("Tried to access config for uninitialised GL context")
        }

        val MAX_TEXTURE_UNITS: Int get() = CONFIG.get().maxTextureUnits
    }

}
